app/src/main/java/com/waseefakhtar/doseapp/feature/addmedication/AddMedicationRoute.kt

```diff
@@ -98,280 +102,288 @@ fun AddMedicationScreen(
     var isAfternoonSelected by rememberSaveable { mutableStateOf(false) }
     var isEveningSelected by rememberSaveable { mutableStateOf(false) }
     var isNightSelected by rememberSaveable { mutableStateOf(false) }
+    val context = LocalContext.current
 
-    Column(
-        modifier = Modifier
-            .padding(0.dp, 16.dp)
-            .verticalScroll(rememberScrollState()),
-        verticalArrangement = Arrangement.spacedBy(8.dp)
-    ) {
-        Row(
-            horizontalArrangement = Arrangement.spacedBy(16.dp),
-            verticalAlignment = Alignment.CenterVertically
-        ) {
-            FloatingActionButton(
+    Scaffold(
+        topBar = {
+            TopAppBar(
+                modifier = Modifier
+                    .padding(vertical = 8.dp),
+                navigationIcon = {
+                    FloatingActionButton(
+                        onClick = {
+                            analyticsHelper.logEvent(AnalyticsEvents.ADD_MEDICATION_ON_BACK_CLICKED)
+                            onBackClicked()
+                        },
+                        elevation = FloatingActionButtonDefaults.elevation(0.dp, 0.dp)
+                    ) {
+                        Icon(imageVector = Icons.Default.ArrowBack, contentDescription = "Back")
+                    }
+                }, title = {
+                    Text(
+                        text = stringResource(id = R.string.add_medication),
+                        fontWeight = FontWeight.Bold,
+                        style = MaterialTheme.typography.displaySmall
+                    )
+                }
+            )
+        },
+        bottomBar = {
+            Button(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(vertical = 8.dp)
```

Comment:
Can we keep the same vertical padding as the button in MedicationConfirmRoute so they're in the same position?

Reply:
It's the same as before, earlier above the button, we were using a spacer of height 8.dp, since I added the button layout to scaffold's bottomBar, so instead of having the spacer used earlier, I added the padding to the button of same 8.dp